### ISSUE TITLE
Evidence of learning:

### ISSUE THREAD

**Issue by AidanRocke** [AidanRocke/variational_empowerment#9]
1. Is the loss going down?
2. Is the heatmap of the Empowerment critic changing over time?
3. Are the parameters of the source and decoder changing?

**Comment by AidanRocke** [comment]
It appears that the Empowerment critic is definitely changing over time but it's also a good idea to inspect the parameters of the source and decoder. 

Ex:

            
```
            ## check variances:
            
            if count % 1000 == 0:
                for j in range(10):
                    # test the decoder:
                    ss_ = np.concatenate((np.zeros(2),X[j],X[j])).reshape((1,6))
                    mu, log_sigma = A.sess.run([A.decoder_mu,A.decoder_log_sigma], feed_dict={ A.decoder_input_n: ss_})                    
                    actions_[obs][j], variances_[obs][j] = mu[0], np.exp(log_sigma[0])
                    
                    print(mu[0])
                    
                obs += 1
```

**Comment by AidanRocke** [comment]
As shown in the loss analysis experiment with two different random seeds, the loss does go down for both the squared loss and decoder loss. It must be noted that:

1. The squared loss diminishes by a factor of 1e5 whereas the decoder loss, which represents a maximum likelihood problem, goes down by a factor of 3.
2. It's not clear to me why the squared loss gets stuck at 15.00 even after doubling the number of training iterations.

**Comment by AidanRocke** [comment]
As shown in the [gradient norms experiment](https://github.com/AidanRocke/variational_empowerment/blob/master/experiments/gradient_norms.ipynb), the gradient norm for both the source and critic goes down pretty quickly. I suspect that the decoder doesn't improve much in 100 iterations because the actions used are mostly random until we reach 5000 iterations. 

It might be worth just looking at the gradient norm of the decoder for 10k iterations though just to be sure.
